sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobClientBase.java

```diff
@@ -387,6 +392,71 @@ public BlobInputStream openInputStream(BlobInputStreamOptions options, Context c
             }).block();
     }
 
+    /**
+     * Opens a seekable byte channel in read-only mode to download the blob.
+     *
+     * @param options {@link BlobSeekableByteChannelReadOptions}
+     * @param context {@link Context}
+     * @return An <code>InputStream</code> object that represents the stream to use for reading from the blob.
+     * @throws BlobStorageException If a storage service error occurred.
+     */
+    public SeekableByteChannel openSeekableByteChannelRead(BlobSeekableByteChannelReadOptions options,
+        Context context) {
+        context = context == null ? Context.NONE : context;
+        options = options == null ? new BlobSeekableByteChannelReadOptions() : options;
+        ConsistentReadControl consistentReadControl = options.getConsistentReadControl() == null
+            ? ConsistentReadControl.ETAG : options.getConsistentReadControl();
+        int chunkSize = options.getBlockSize() == null ? 4 * Constants.MB : options.getBlockSize();
+        long initialPosition = options.getInitialPosition() == null ? 0 : options.getInitialPosition();
+
+        ByteBuffer initialRange = ByteBuffer.allocate(chunkSize);
+        BlobProperties properties;
+        try (ByteBufferBackedOutputStream dstStream = new ByteBufferBackedOutputStream(initialRange)) {
+            BlobDownloadResponse response = this.downloadStreamWithResponse(dstStream,
+                new BlobRange(initialPosition, (long) initialRange.remaining()), null /*downloadRetryOptions*/,
+                options.getRequestConditions(), false, null, context);
+            properties = ModelHelper.buildBlobPropertiesResponse(response).getValue();
+        } catch (IOException e) {
+            throw LOGGER.logExceptionAsError(new RuntimeException(e));
+        }
+
+        initialRange.limit(initialRange.position());
+        initialRange.rewind();
+
+        BlobClientBase behaviorClient = this;
+        BlobRequestConditions requestConditions = options.getRequestConditions();
+        switch (consistentReadControl) {
+            case NONE:
+                break;
+            case ETAG:
+                requestConditions = requestConditions != null ? requestConditions : new BlobRequestConditions();
+                // If etag locking but no explicitly specified etag, use the etag from prefetch
+                if (requestConditions.getIfMatch() == null) {
+                    requestConditions.setIfMatch(properties.getETag());
+                }
+                break;
+            case VERSION_ID:
+                if (properties.getVersionId() == null) {
+                    throw LOGGER.logExceptionAsError(
+                        new UnsupportedOperationException("Versioning is not supported on this account."));
```

Comment:
Would include in the error message that version ID locking cannot be supported since versioning isn't supported on the account, to clarify why the error happened.